src/main/java/uk/ac/ebi/ampt2d/accession/SHA1AccessionGenerator.java

```diff
@@ -23,16 +23,10 @@
 
 public class SHA1AccessionGenerator<T> extends SingleAccessionGenerator<T, String> {
 
-    private String namespace;
-
-    public SHA1AccessionGenerator(String namespace) {
-        this.namespace = namespace;
-    }
-
     @Override
     protected String generateAccession(T object) {
         byte[] hashBytes = ByteBuffer.allocate(4).putInt(object.hashCode()).array();
```

Comment:
I'm not particulary happy with this, I think it is better if we make T implement an interface and add a specific method or use json form of the class, @cyenyxe thoughts?

Reply:
@jorizci - Do you mean to define the class like  SHA1AccessionGenerator<T extends AccessioningObject> ?